### PR TITLE
Add metric for failed memtable flushes

### LINKED ISSUE #3316: Add metric for failed memtable flush
*Installation details*
Scylla version (or git commit hash): >= 1.0

We don't account for failed memtable flushes. We assume errors in flushing are transitional, but that may not be the case, and keep retrying ad-infinitum. Having a counter of failed writes could provide the user with visibility that a node is having problems.

### PR DESCRIPTION
**Pull request by kbr-**
Resolves #3316.

### CODE REVIEW

**Comment by kbr-** [comment]
@psarna @avikivity could you review please?

File: database.cc [hunk 1]
```diff
@@ -327,6 +327,9 @@ database::setup_metrics() {
         sm::make_gauge("pending_flushes_bytes", _cf_stats.pending_memtables_flushes_bytes,
                        sm::description("Holds the current number of bytes in memtables that are currently being flushed to sstables. "
                                        "High value in this metric may be an indication of storage being a bottleneck.")),
+        sm::make_gauge("failed_flushes", _cf_stats.failed_memtables_flushes_count,
+                       sm::description("Holds the number of failed memtable flushes. "
+                                       "High value in this metric may indicate a permament failure to flush a memtable.")),
```
Comment (psarna) [inline database.cc]: perma**n**ent

Comment (kbr-) [reply database.cc]: Done

**Comment by kbr-** [comment]
@penberg @nyh could you review this please?

**Comment by avikivity** [comment]
queued as d6736a304a3e87a917aaa5b5fed60a18b407464f

**Pull request closed without merging**